src/Week2/LogAnalysis.hs

```diff
@@ -113,6 +131,18 @@ inOrder Leaf                  = []
 inOrder (Node left msg right) = inOrder left ++ [msg] ++ inOrder right
 
 
+-- The above can also be rewritten as follows
+-- This has an advantage of performace over doing list concatenation recursively
+inOrder' :: MessageTree -> [LogMessage]
+inOrder' Leaf = []
+inOrder' n = inOrderUtil n []
+  where
+    inOrderUtil z xs =
+      case z of
+        Leaf -> xs
+        (Node left root right) -> inOrderUtil left (root : (inOrderUtil right xs))
```

Comment:
Just a few pointers on style:
- You don't need to use the parentheses around `Node left root right`
- You can use the `$` operator to eliminate the second set of parentheses on the right